src/error.rs

```diff
@@ -45,7 +45,7 @@ pub enum MoveError {
     PerpetualCheckLose,
     EnemysTurn,
     NonMovablePiece,
-    Inconsistent,
+    Inconsistent(&'static str),
```

Comment:
Added a message (reason) because it was difficult to understand what's happening

Reply:
LGTM, it'd make life a little easier ;)